target/linux/ar71xx/image/tp-link.mk

```diff
@@ -757,7 +757,25 @@ define Device/tl-wr1043nd-v3
     DEVICE_PROFILE := TLWR1043
     TPLINK_HWID := 0x10430003
 endef
-TARGET_DEVICES += tl-wr1043nd-v1 tl-wr1043nd-v2 tl-wr1043nd-v3
+
+define Device/tl-wr1043nd-v4
+    $(Device/tplink)
+    DEVICE_TITLE := TP-LINK TL-WR1043N/ND v4
+    DEVICE_PACKAGES := kmod-usb-core kmod-usb2 kmod-usb-ledtrig-usbport
+    BOARDNAME := TL-WR1043ND-v4
+    DEVICE_PROFILE := TLWR1043
+    TPLINK_HWID := 0
+    TPLINK_HWREV := 0
```

Comment:
I have one more request. Could you try setting TPLINK_HWID to 0x10430004 and remove the TPLINK_HWREV definition (so it defaults to 1), and check if the stock firmware still accepts the image?

The stock firmware images contain 0 here, but I suspect that the upgrader doesn't check it at all (as the mktplinkfw image is embedded into the safeloader image). Setting TPLINK_HWID to a unique value would allow the LEDE sysupgrader to check for the correct model by comparing the values in flash and in the image.

(this might become redundant soon with the new LEDE image metadata, but we don't have that for ar71xx yet)